src/backend/looped/self/part_water.cpp

```diff
@@ -0,0 +1,30 @@
+#include "backend/looped/looped.hpp"
+#include "natives.hpp"
+#include "backend/looped_command.hpp"
+
+namespace big
+{
+	class partwater : looped_command
+	{
+		using looped_command::looped_command;
+
+		virtual void on_tick() override
+		{
+			float offset[] = { -4, 4 };
+			for (int i = 0; i < 5; i++)
+			{
+				if (i < 2)
+				{
+					self::pos.x += offset[(i % 2 == 0)];
+				}
+				else if (i < 4)
+				{
+					self::pos.y += offset[(i % 2 == 0)];
+					WATER::MODIFY_WATER(self::pos.x, self::pos.y, -500000.0f, 0.2f);
+				}
+			}
+		}
```

Comment:
I don't follow what's going on here. The loop unrolls to this:

```
self::pos.x += 4;

self::pos.x += -4;

self::pos.y += 4;
WATER::MODIFY_WATER(self::pos.x, self::pos.y, -500000.0f, 0.2f);

self::pos.y += -4;
WATER::MODIFY_WATER(self::pos.x, self::pos.y, -500000.0f, 0.2f);
```

- Mutating `self::pos` is weird, you probably want to make a locally-scoped copy and mutate it
- The first two loop iterations do nothing
- The last loop iteration does nothing

Reply:
Awesome catch the native call was in the wrong place, was working too fast.